Sources/SQLite/Typed/Coding.swift

```diff
@@ -0,0 +1,347 @@
+                self.encoder.setters.append(Expression(key.stringValue) <- data)
+            }
+            else {
+                let encoded = try JSONEncoder().encode(value)
+                let string = String(data: encoded, encoding: .utf8)
+                self.encoder.setters.append(Expression(key.stringValue) <- string)
+            }
+        }
+
+        func encode(_ value: Int8, forKey key: Key) throws {
+            throw EncodingError(description: "encoding an Int8 is not supported")
+        }
+
+        func encode(_ value: Int16, forKey key: Key) throws {
+            throw EncodingError(description: "encoding an Int16 is not supported")
+        }
+
+        func encode(_ value: Int32, forKey key: Key) throws {
+            throw EncodingError(description: "encoding an Int32 is not supported")
+        }
+
+        func encode(_ value: Int64, forKey key: Key) throws {
+            throw EncodingError(description: "encoding an Int64 is not supported")
+        }
+
+        func encode(_ value: UInt, forKey key: Key) throws {
+            throw EncodingError(description: "encoding an UInt is not supported")
+        }
+
+        func encode(_ value: UInt8, forKey key: Key) throws {
+            throw EncodingError(description: "encoding an UInt8 is not supported")
+        }
+
+        func encode(_ value: UInt16, forKey key: Key) throws {
+            throw EncodingError(description: "encoding an UInt16 is not supported")
+        }
+
+        func encode(_ value: UInt32, forKey key: Key) throws {
+            throw EncodingError(description: "encoding an UInt32 is not supported")
+        }
+
+        func encode(_ value: UInt64, forKey key: Key) throws {
+            throw EncodingError(description: "encoding an UInt64 is not supported")
+        }
+
+        func nestedContainer<NestedKey>(keyedBy keyType: NestedKey.Type, forKey key: Key) -> KeyedEncodingContainer<NestedKey> where NestedKey : CodingKey {
+            fatalError("encoding a nested container is not supported")
+        }
+
+        func nestedUnkeyedContainer(forKey key: Key) -> UnkeyedEncodingContainer {
+            fatalError("encoding nested values is not supported")
+        }
+    }
+
+    fileprivate var setters: [SQLite.Setter] = []
+    let codingPath: [CodingKey] = []
+    let userInfo: [CodingUserInfoKey: Any]
+
+    init(userInfo: [CodingUserInfoKey: Any]) {
+        self.userInfo = userInfo
+    }
+
+    func singleValueContainer() -> SingleValueEncodingContainer {
+        fatalError("not supported")
+    }
+
+    func unkeyedContainer() -> UnkeyedEncodingContainer {
+        fatalError("not supported")
+    }
+
+    func container<Key>(keyedBy type: Key.Type) -> KeyedEncodingContainer<Key> where Key : CodingKey {
+        return KeyedEncodingContainer(SQLiteKeyedEncodingContainer(encoder: self))
+    }
+}
+
+fileprivate class SQLiteDecoder : Decoder {
+    struct DecodingError : Error, CustomStringConvertible {
```

Comment:
Yup, I didn't realize those existed. But I am leaving them as thrown errors instead of fatal errors because I would rather that be caught than bring down the whole program.